src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -267,462 +263,298 @@ void GetPIUs(EnergyPlusData &state)
+                    ErrorsFound = true;
+                }
 
-        // Add fan to component sets array
-        SetUpCompSets(state,
-                      thisPIU.UnitType,
-                      thisPIU.Name,
-                      "UNDEFINED",
-                      state.dataIPShortCut->cAlphaArgs(8),
-                      state.dataIPShortCut->cAlphaArgs(4),
-                      "UNDEFINED");
-
-        // Add reheat coil to component sets array
-        SetUpCompSets(state,
-                      thisPIU.UnitType,
-                      thisPIU.Name,
-                      state.dataIPShortCut->cAlphaArgs(9),
-                      state.dataIPShortCut->cAlphaArgs(10),
-                      state.dataIPShortCut->cAlphaArgs(6),
-                      state.dataIPShortCut->cAlphaArgs(5));
-
-        // Register component set data
-        TestCompSet(state,
-                    thisPIU.UnitType,
-                    thisPIU.Name,
-                    state.dataLoopNodes->NodeID(thisPIU.PriAirInNode),
-                    state.dataLoopNodes->NodeID(thisPIU.OutAirNode),
-                    "Air Nodes");
-
-        for (int ADUNum = 1; ADUNum <= (int)state.dataDefineEquipment->AirDistUnit.size(); ++ADUNum) {
-            if (thisPIU.OutAirNode == state.dataDefineEquipment->AirDistUnit(ADUNum).OutletNodeNum) {
-                //      AirDistUnit(ADUNum)%InletNodeNum = PIU(PIUNum)%InletNodeNum
-                state.dataDefineEquipment->AirDistUnit(ADUNum).InletNodeNum = thisPIU.PriAirInNode;
-                thisPIU.ADUNum = ADUNum;
-            }
-        }
-        // one assumes if there isn't one assigned, it's an error?
-        if (thisPIU.ADUNum == 0) {
-            ShowSevereError(state, format("{}No matching Air Distribution Unit, for PIU = [{},{}].", RoutineName, thisPIU.UnitType, thisPIU.Name));
-            ShowContinueError(state, format("...should have outlet node = {}", state.dataLoopNodes->NodeID(thisPIU.OutAirNode)));
-            ErrorsFound = true;
-        } else {
+                std::string const heating_control_type = ip->getAlphaFieldValue(fields, objectSchemaProps, "heating_control_type");
+                thisPIU.heatingControlType = HeatCntrlBehaviorType::Invalid;
+                if (thisPIU.fanControlType == FanCntrlType::VariableSpeedFan) {
+                    if (Util::SameString(heating_control_type, "Staged")) {
+                        thisPIU.heatingControlType = HeatCntrlBehaviorType::StagedHeaterBehavior;
+                    } else if (Util::SameString(heating_control_type, "Modulated")) {
+                        thisPIU.heatingControlType = HeatCntrlBehaviorType::ModulatedHeaterBehavior;
+                    } else {
+                        ShowSevereError(state, format("Illegal Heating Control Type = {}", heating_control_type));
+                        ShowContinueError(state, format("Occurs in {} = {}", cCurrentModuleObject, thisPIU.Name));
```

Comment:
Specifically, the thing I was trying to test was this error message.

The "Heating Control Type" field isn't marked required, because it's used only when "Fan Control Type" = VariableSpeed.

The error is less than informative in case you just forgot to fill it.

```
   ** Severe  ** Illegal Heating Control Type = 
   **   ~~~   ** Occurs in AirTerminal:SingleDuct:SeriesPIU:Reheat = SPACE1-1 VAV REHEAT
   **  Fatal  ** GetPIUs:  Errors found in getting input.  Preceding conditions cause termination.
```


```diff
--- a/testfiles/5ZoneAirCooledConvCoef_VSFan.idf
+++ b/testfiles/5ZoneAirCooledConvCoef_VSFan.idf
@@ -2436,7 +2436,7 @@
     0.0001,                  !- Convergence Tolerance
     VariableSpeed,
     0.5,
-    Modulated,
+    ,
     ,
     ;
```

Reply:
Thanks for pointing it out, I'll address it today.